src/gov/nist/javax/sip/stack/SIPDialog.java

```diff
@@ -4409,8 +4409,13 @@ public void setEarlyDialogTimeoutSeconds(int seconds) {
     public void checkRetransmissionForForking(SIPResponse response) {
         final int statusCode = response.getStatusCode();
         final String responseMethod = response.getCSeqHeader().getMethod();
-        final long responseCSeqNumber = response.getCSeq().getSeqNumber();   
-        boolean isRetransmission = !responsesReceivedInForkingCase.add(statusCode + "/" + responseCSeqNumber + "/" + responseMethod);            
+        final long responseCSeqNumber = response.getCSeq().getSeqNumber();
+        RSeq rseq = (RSeq) response.getHeader(RSeqHeader.NAME);
+        String responseRSeqNumber = "";
+        if(rseq != null) {
+            responseRSeqNumber = "/" + rseq.getSeqNumber();
+        }
+        boolean isRetransmission = !responsesReceivedInForkingCase.add(statusCode + "/" + responseCSeqNumber + "/" + responseMethod + responseRSeqNumber);
```

Comment:
I assume if no RSeq header is present, the logic remains as previously,right?

Have you tested this somehow? Any possibility to provide automated junit test?

Ideally these logic should be targeted only for prov resps (1XX) and hopefully enabled form stack conf flag...